pandas_gbq/schema/pandas_to_bigquery.py

```diff
@@ -188,29 +197,33 @@ def value_to_bigquery_field(name, value) -> Optional[schema.SchemaField]:
     return None
 
 
-def values_to_bigquery_field(name, values) -> Optional[schema.SchemaField]:
+def values_to_bigquery_field(
+    name, values, default_type="STRING"
+) -> Optional[schema.SchemaField]:
     value = pandas_gbq.core.pandas.first_valid(values)
 
-    # All NULL, type not determinable.
+    # All values came back as NULL, thus type not determinable by this method.
```

Comment:
I know we are pushing to merge to enable further work, so the following is merely a #PREFERENCE or #SUGGESTION, not a blocker.

* The code here works and tests pass so I don't feel like forcing us to fix something that isn't broken, but I wonder if we could clean up the code slightly (better readability, maintainability, efficiency). 
  * we use `isinstance(v, str)` in two functions despite the fact that the outer function calls the inner function, which does a string check and then the outer function immediately does a second `isinstance()` check.
  * we use a `field is not None` double negative style check when I suspect we can safely do an `if field` check.

Will leave it up to you to decide whether any of these changes are reasonable in this PR.

```
    value = pandas_gbq.core.pandas.first_valid(values)

    # All values came back as NULL, thus type not determinable by this method.
    # Return None so we can try other methods.
    if value is None:
        return None

    field = value_to_bigquery_field(name, value, default_type=default_type)
    if field:
        return field

    # Check plain ARRAY values. Exclude mapping types to allow pyarrow to determine
    # STRUCT which can examine more values to determine all the keys
    if isinstance(value, collections.abc.Iterable) and not isinstance(
        value, collections.abc.Mapping
    ):
        # This collection may contain only None elements or is empty, so get the
        # first non-None value we can find.
        valid_item = pandas_gbq.core.pandas.first_array_valid(values)
        field = value_to_bigquery_field(name, valid_item, default_type=default_type)
        if field:
            return schema.SchemaField(name, field.field_type, mode="REPEATED")

    return None
```

Reply:
Good point. Removed the redundant string check.